### PR TITLE
Add vagrant setup

### LINKED ISSUE #21: Create vagrant file
A Vagrant file would be nice to develop the software without having to go through all the steps of manual setup. It would be much easier to just use `vagrant up`.

### PR DESCRIPTION
**Pull request by sirkrypt0**
This adds all files needed to setup the local environment using Vagrant.

Closes #21

### CODE REVIEW

**Review by frcroth — APPROVED** [review]

**Review by felixauringer — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -17,6 +17,14 @@ Created in the [Software Engineering II course](https://hpi.de/plattner/teaching
 
 ## Development Setup
 
+### Vagrant
+* To create the machine, run `vagrant up`.
+* Enter the machine using `vagrant ssh`.
+* Go to the project directory `cd hpi-swt2`.
+* Finally start the rails dev server `rails s -b 0.0.0.0` (`-b 0.0.0.0` makes sure that you can reach the dev server locally).
```
Comment (chrisma) [inline README.md]:
```suggestion
* Start the dev server `rails s -b 0.0.0.0` (`-b ` / `--binding=IP` allows connections from outside the VM).
```

File: README.md [hunk 2]
```diff
@@ -17,6 +17,14 @@ Created in the [Software Engineering II course](https://hpi.de/plattner/teaching
 
 ## Development Setup
 
+### Vagrant
+* To create the machine, run `vagrant up`.
```
Comment (chrisma) [inline README.md]:
```suggestion
* To [create the VM](https://www.vagrantup.com/docs/cli/up), run `vagrant up` (based on the `Vagrantfile`)`.
```

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure("2") do |config|
+  config.vm.box = "ubuntu/bionic64"
+  # config.vm.box_url = "https://github.com/hpi-swt2/swt2-vagrant/releases/download/v0.2/swt2-trusty64.box"
+
+  # use 32bit machine
+  #config.vm.box = "swt2/trusty32"
+  #config.vm.box_url = "https://github.com/hpi-swt2/swt2-vagrant/releases/download/v0.2/swt2-trusty32.box"
```
Comment (chrisma) [inline Vagrantfile]: This can go, right?

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  # previously there were performance issues. Try to use the following optimizations
+  # disable if problems occur or fall back to the 32bit VM
+  config.vm.provider :virtualbox do |vb|
+    vb.customize ["modifyvm", :id, "--memory", "2048"]
+    vb.customize ["modifyvm", :id, "--cpus", 2]
+    vb.customize ["modifyvm", :id, "--ioapic", "on"]
+    vb.customize ["modifyvm", :id, "--hwvirtex", "on"]
+  end
```
Comment (chrisma) [inline Vagrantfile]: I'm curious, have you tried whether these settings still make a difference?

File: provision.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+eval "$(rbenv init -)"
+
+cd ~/hpi-swt2
+
+# Install ruby-build
+mkdir -p "$(rbenv root)"/plugins
+git clone https://github.com/rbenv/ruby-build.git "$(rbenv root)"/plugins/ruby-build
+
+# Install ruby version
+rbenv install
```
Comment (chrisma) [inline provision.sh:22]:
```suggestion
rbenv install 2.7.2
```

**Review by chrisma — APPROVED** [review]
Nice! Thanks!

File: provision.sh (hunk 5 above)
Comment (chrisma) [reply provision.sh:22]: Or does `rbenv install` read the `.ruby-version` file by default?

File: Vagrantfile (hunk 4 above)
Comment (sirkrypt0) [reply Vagrantfile]: I tested it and couldn't spot any difference without them, so I removed those.

**Comment by chrisma** [comment]
Awesome, thanks!

**Pull request merged**